rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -141,6 +166,7 @@ impl Expr {
             Expr::Column(name) => Ok(schema.field_with_name(name)?.data_type().clone()),
             Expr::ScalarVariable(_) => Ok(DataType::Utf8),
             Expr::Literal(l) => Ok(l.get_datatype()),
+            Expr::Case { when_then_expr, .. } => when_then_expr[0].1.get_type(schema),
```

Comment:
I feel like there should be a check somewhere that all the `then` and `else` clauses result in the same type -- however, I suppose we would just get an arrow error at runtime, which while not ideal, could easily be fixed subsequently.

Reply:
I agree. I can look at adding this tomorrow.

Reply:
I've added a check for the case where all "then" values are literals, to make sure they have the same type.